src/components/common/TmSessionHardware.vue

```diff
@@ -58,12 +58,19 @@ export default {
       this.connectionError = error
     },
     async connectLedger() {
-      this.setStatus(`detect`)
-      try {
-        await this.$store.dispatch(`connectLedgerApp`)
-      } catch (error) {
-        this.setStatus(`connect`)
-        this.setConnectionError(error.message)
+      if (
+        navigator.userAgent.includes(`Chrome`) ||
+        navigator.userAgent.includes(`Opera`)
+      ) {
+        this.setStatus(`detect`)
+        try {
+          await this.$store.dispatch(`connectLedgerApp`)
+        } catch (error) {
+          this.setStatus(`connect`)
+          this.setConnectionError(error.message)
+        }
+      } else {
+        this.setConnectionError(`Please use Chrome, Brave, or Opera`)
```

Comment:
Please add the warning right away. So when the user using Firefox visits the page, he sees the message. Then also disable the "Submit"/"Sign In" button in that case.
The thinking: right now the user expects everything to work. then he presses a button expecting to get signed in and then is sad as it doesn't work. where if we show the message and disable the button, the user never creates an expectation.